### ISSUE TITLE
Not sure how to generate mels to be used with a pre-trained wavenet vocoder.

### ISSUE THREAD

**Issue by deli666** [NVIDIA/tacotron2#50]
I put the inference.ipynb code into inference.py code so that it can run on gpu and save plots, audio (Griffin_lim) and mel_spectrgram.
The plots and audio are good, but when I put the generated Mel into pretrained wavenet_vocoder(https://github.com/r9y9/wavenet_vocoder) , the audio generated is not reasonable.
I wonder if this way of save mel-spectrogram is wrong.

 mel_outputs, mel_outputs_postnet, _, alignments = model.inference(sequence)
 np.save("test_mel.npy", mel_outputs.data.cpu().numpy()[0].T, allow_pickle=False)

**Comment by rafaelvalle** [comment]
Save mel_outputs_postnet, not mel_outputs. 
Then you must process the mel-spectrogram such that it is the same as the one used to train your pre-trained model. 

The code for pre-processing mels for Ryuchi's wavenet is in this issue:
https://github.com/NVIDIA/tacotron2/issues/40

**Comment by deli666** [comment]
@rafaelvalle Thank you so much for help!
I am using LJSpeech data to train this model (97k steps) and WaveNet_Vocoder(https://github.com/r9y9/wavenet_vocoder) (620k steps)

And here is the sound generated: 
Text: "This is an example of the sound generated. It may sound awful, but it is just a start"
[GriffinLim.zip](https://github.com/NVIDIA/tacotron2/files/2144155/GriffinLim.zip)
[WaveNet.zip](https://github.com/NVIDIA/tacotron2/files/2144158/WaveNet.zip)


There is still some noise in the audio and not smooth enough. After how many steps of training I can generate audio like demo.wav?

Besides, I plan to train this model with my own mandarin dataset. Do you think there should be some hparams being changed?

**Comment by rafaelvalle** [comment]
Thanks for sharing these, @deli666.
Did you train the wavenet vocoder yourself or did you use the wavenet vocoder on https://github.com/r9y9/wavenet_vocoder?

**Comment by deli666** [comment]
@rafaelvalle I trained the wavenet vocoder myself.

**Comment by rafaelvalle** [comment]
Try using the pre trained vocoder.

**Comment by deli666** [comment]
@rafaelvalle This is the sound generated with pre-trained vocoder at https://github.com/r9y9/wavenet_vocoder

It sounds much smoother. 

[sample_audio.zip](https://github.com/NVIDIA/tacotron2/files/2148860/sample_audio.zip)

**Comment by rafaelvalle** [comment]
That's a good start! You probably need to train Tacotron 2 longer, eventually decreasing the learning rate!

**Comment by NewEricWang** [comment]
@deli666 , Could you share the code that save the mel_spectrgram generated by Tacotron2 and convert the mel_spectrgram into the mel-spectrogram for the pre-trained wavenet vocoder at https://github.com/r9y9/wavenet_vocoder.  I have tried to do it, but I failed to get good results. Thank you!

**Comment by deli666** [comment]
@NewEricWang 
[synthesize.zip](https://github.com/NVIDIA/tacotron2/files/2162678/synthesize.zip)
This is my code for generate mel-spectrogram.
Beside, you need to add methods below in wavenet_vocoder when read in the mel.npy

# methods from from Ryuchi's repo
mel = audio._amp_to_db(mel) - hparams.ref_level_db
if not hparams.allow_clipping_in_normalization:
    assert mel.max() <= 0 and mel.min() - hparams.min_level_db >= 0
mel = audio._normalize(mel)

**Comment by deli666** [comment]
@rafaelvalle 
Thanks for reply!
I have trained 500 epochs and the program stopped. I restarted it with 
python train.py --output_directory=outdir --log_directory=logdir
But received some error.
Should I add some argument or change something in code if I want to continue to train?

**Comment by rafaelvalle** [comment]
@deli666 is this the full stack trace? if not can you please post the full stack trace?
